packages/org.json_schema.contrib/internal/TypesGenerator.pkl

```diff
@@ -669,5 +669,100 @@ local function generateBaseType(schema: JsonSchema, typeNames: Type.TypeNames):
     Pair(schema, generateListingType(schema, typeNames))
   else if (isMappingSchema(schema))
     Pair(schema, generateObjectType(schema, typeNames))
+  else if (schema.allOf != null)
+    let (collatedSchema = collateAllOf(schema))
+      generateBaseType(collatedSchema, typeNames)
   else
     Pair(schema, utils.declaredType("Any"))
+
+local function allOfErr(msg: String) = throw("Unable to combine allOf elements into one schema: \(msg)")
+
+function collateAllOf(schema: JsonSchema(allOf != null)): JsonSchema =
+  schema.allOf.fold(schema, (res, rawElem) ->
+    let (resolvedElem = if (rawElem is JsonSchema && rawElem.$$refUri != null) ref.resolveRef(baseUri, rawElem) else rawElem)
+      let (elem = if (resolvedElem is JsonSchema && resolvedElem.allOf != null) collateAllOf(resolvedElem) else resolvedElem)
+        new JsonSchema {
+          // core
+          type = if (res.type != null && elem.type != null && res.type != elem.type) allOfErr("conflicting type")
+          else elem.type ?? res.type
+          `const` = if (res.`const` != null && elem.`const` != null && res.`const` != elem.`const`) allOfErr("conflicting const")
+          else elem.`const` ?? res.`const`
+
+          // number
+          multipleOf = if (elem.multipleOf != null && res.multipleOf != null)
+            if (elem.multipleOf is Int && res.multipleOf is Int) math.lcm(elem.multipleOf!! as Int, res.multipleOf!! as Int)
+            else allOfErr("multiple non-integer multipleOf")
+          else elem.multipleOf ?? res.multipleOf
+          minimum = collateMin(elem.minimum, res.minimum)
+          exclusiveMinimum = collateMin(elem.exclusiveMinimum, res.exclusiveMinimum)
+          maximum = collateMax(elem.maximum, res.maximum)
+          exclusiveMaximum = collateMax(elem.exclusiveMaximum, res.exclusiveMaximum)
+
+          // string
+          pattern = if (elem.pattern != null && res.pattern != null) "\(elem.pattern)|\(res.pattern)"
```

Comment:
Probably best to wrap each alternative in a non-capture group so each sub-pattern is treated as its own alternative.

```suggestion
          pattern = if (elem.pattern != null && res.pattern != null) "(?:\(elem.pattern))|(?:\(res.pattern))"
```